### ISSUE TITLE
keyboard issue when switching between TextInput widgets on android

### ISSUE THREAD

**Issue by hussam132** [kivy/kivy#7698]
**Software Versions**
* Python:3.7
* OS:  android 11
* Kivy: 2.0.0
* Kivy installation method:
using python for android

**Describe the bug**
when we have more then on `TextInput` in same screen on android the keyboard some times not appears when  switching directly from `TextInput` to another only work fine if I close the keyboard and focus the other one 

**Code and Logs and screenshots**
```python
from kivy.core.window import Window
from kivy.lang import Builder
from kivymd.app import MDApp

kv = '''
BoxLayout:
    orientation:'vertical'
    spacing:'20dp'
    padding:'20dp'

    TextInput
    TextInput
    TextInput
    TextInput
'''


class TestApp(MDApp):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        Window.keyboard_anim_args = {"d": 0.2, "t": "linear"}
        Window.softinput_mode = "below_target"

    def build(self):
        return Builder.load_string(kv)


TestApp().run()

```

**Additional context**

the video will show the issue on android

https://user-images.githubusercontent.com/52833102/142660765-840414b5-e552-4deb-936a-ea6d5ab8d606.mp4

**Comment by HyTurtle** [comment]
https://github.com/kivy/kivy/blob/master/kivy/uix/behaviors/focus.py#L400

Can you confirm the *issue* is here? Would explain why (prior to example above) it wasn't replicated.

**Comment by FilipeMarch** [comment]
> https://github.com/kivy/kivy/blob/master/kivy/uix/behaviors/focus.py#L400
> 
> Can you confirm the *issue* is here? Would explain why (prior to example above) it wasn't replicated. Or were you seeing the same prior to setting keyboard_mode to managed?

Yes, it is happening due to `_on_focus` being called by two `TextInput`s in an unorganized and non-deterministic manner. I was printing who was executing these functions first, and it was a complete mess!

Sometimes T2 would say "I will create keyboard now", and afterwards T1 would say "I will destroy keyboard now", but the order should always be 1) The keyboard disappears; 2) The keyboard appears.

 But `Eventloop.window.request_keyboard` has a callback that executes `self.focus = False` (imagine what 'auto' will do when this happens), so whenever the android keyboard was requested by one textfield (the smartphone blinks, closing the previous keyboard and adding a new one instantly), the other textfield changes focus afterwards, due to `Event.loop` callback and it kills the android keyboard. 

I have never used keyboard_mode 'manage' before. I didn't even knew it existed. During this night I was able to reproduce the issue both on computer and smartphone deterministically, and the code I posted prevented the bug on both. 

Sorry if my explanation is not the best, but using the debugger makes it clear what is really happening there, probably there is a better fix but I will use this workaround ('managed') for now

**Comment by HyTurtle** [comment]
Not seeing where your debug prints were sort of limits the usefulness; it probably is somewhat a race condition. Did you try with `0` there instead of `.1`, as in schedule it for the next clock tick. In which case probably no harm PR'ing that...

On the other hand/further thinking; if that retains the issue then presumably due to it being a property change; on two widgets both being propagated on next tick (which I believe is the case now, rather than an immediate firing) with the order being unspecified.

**Comment by FilipeMarch** [comment]
I've tested with `Clock.schedule_once(self.create_keyboard)`, but then the android keyboard disappears when you click on two textfields consecutively. `Clock.schedule_once(self.create_keyboard, .05)` still had the same problem, and `Clock.schedule_once(self.create_keyboard, .075)` works fine, `Clock.schedule_once(self.create_keyboard, .1)` seems safer and nice

**Comment by antorix** [comment]
I found a very similar bug, but it's completely gone when disabling "Flow" on the Gboard keyboard. This bug appears only with "Flow" on.
